suministrospr/suministros/management/commands/import_data.py

```diff
@@ -16,31 +17,68 @@ def normalize(self, text):
     def add_arguments(self, parser):
         parser.add_argument("target_folder", type=str)
 
+    def get_suministro(self, slug):
+        try:
+            return Suministro.objects.get(slug=slug)
+        except Suministro.DoesNotExist:
+            return None
+
+    def get_import_data(self, data):
+        municipality = self.normalize(data.get("municipio", ""))
+        title = data.get("title", "")
+        slug = (
+            data.get("url")
+            .replace("https://suministrospr.com/sectores", "")
+            .replace("/", "")
+        )
+        return {
+            "title": title,
+            "content": data.get("content", ""),
+            "municipality": municipality,
+            "slug": slug,
+        }
+
+    def get_form(self, import_data, suministro):
+        if suministro:
+            return SuministroModelForm(import_data, instance=suministro)
+        else:
+            return SuministroModelForm(import_data)
+
     def handle(self, *args, **options):
         target_folder = options["target_folder"]
 
         file_paths = glob.glob(f"{target_folder}/*.json")
 
         for file_path in file_paths:
-            self.stdout.write(f"Importing file {file_path}")
+            self.stdout.write(f"=> Importing file {file_path}")
             try:
                 with open(file_path, "r") as json_file:
                     data = json.load(json_file)
                     for item in data:
-                        municipality = self.normalize(item.get("municipio", ""))
-                        import_data = {
-                            "title": item.get("title", ""),
-                            "content": item.get("content", ""),
-                            "municipality": municipality,
-                        }
-                        suministro = SuministroModelForm(import_data)
-
-                        if suministro.is_valid():
+                        import_data = self.get_import_data(item)
+                        slug = import_data.get("slug")
+                        existing_suministro = self.get_suministro(slug)
+
+                        suministro_form = self.get_form(
+                            import_data, existing_suministro
+                        )
+
+                        if suministro_form.is_valid():
+                            suministro = suministro_form.save(commit=False)
+                            suministro.slug = slug
```

Comment:
@froi here we make sure we set an explicit slug instead of generating a new one.

This required setting `overwrite_on_add=False` to our `Suministro.slug` field